src/libDirectoryService/DirectoryService.cpp

```diff
@@ -488,6 +489,38 @@ bool DirectoryService::FinishRejoinAsDS() {
   }
 
   LOG_MARKER();
+  for (auto& i : *m_mediator.m_DSCommittee) {
```

Comment:
Do you need to acquire `m_mediator.m_mutexDSCommittee` here?
Because later on during call to `UpdateDSGuardIdentity` you also acquire it.